File: 0961-n-repeated-element-in-size-2n-array/0961-n-repeated-element-in-size-2n-array.py
class Solution(object):
    def repeatedNTimes(self, nums):
        n = len(nums) // 2
        result = -1

        freq = {}
        for num in nums:
            freq[num] = freq.get(num, 0) + 1
            if freq[num] == n:
                result = num

        return result    

File: 0961-n-repeated-element-in-size-2n-array/0961-n-repeated-element-in-size-2n-array.cpp
class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        int size = nums.size();
        int n = size/2;

        unordered_map <int,int> freq;
        for(int num:nums){
            freq[num]++;
        }
        int result;
        for(auto it:freq){
            if(it.second == n){
                result = it.first;
            }
        }
        return result;
    }
};